extensions/guacamole-auth-json/src/test/java/org/apache/guacamole/auth/json/RequestValidationServiceTest.java

```diff
@@ -0,0 +1,375 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.json;
+
+import java.io.BufferedReader;
+import java.security.Principal;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import javax.servlet.ServletInputStream;
+import javax.servlet.RequestDispatcher;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ * Unit test for RequestValidationService. Verifies that only requests
+ * from trusted hosts are allowed to authenticate.
+ */
+public class RequestValidationServiceTest {
+
+    private class MockConfigurationService extends ConfigurationService {
+
+        private Collection<String> trustedNetworks;
+
+        public MockConfigurationService() {
+            trustedNetworks = Collections.<String>emptyList();
+        }
+
+        public MockConfigurationService(String trustedNetworks) {
+            this.trustedNetworks = Arrays.asList(Pattern.compile(",\\s*").split(trustedNetworks));
+        }
```

Comment:
I'm OK with this as-is, but I suggest instead covering both these cases with just: `String... trustedNetworks`.

Reply:
Could you please be a little more specific. Do you mean that I should make use of just a single constructor?

Reply:
Yep. Reading through this, it occurred to me that what you have here is equivalent to:

```java
public MockConfigurationService(String... trustedNetworks) {
    this.trustedNetworks = Arrays.asList(trustedNetworks);
}
```

with the added benefit that you don't need to rely on string splitting.

Reply:
> Yep. Reading through this, it occurred to me that what you have here is equivalent to:
> 
> ```java
> public MockConfigurationService(String... trustedNetworks) {
>     this.trustedNetworks = Arrays.asList(trustedNetworks);
> }
> ```
> 
> with the added benefit that you don't need to rely on string splitting.

Unfortunately, I cannot confirm that.